election-ui-javascript/src/dataFetcher/index.test.js

```diff
@@ -10,12 +10,20 @@ const mockFakeApi = () => {
       results: [
         {
           'party': 'Independent',
-          'candidate': 'Lord Buckethead',
-          'votes': '9900'
+          'votes': '9900',
+          'candidateId': 1
         }
       ]
     })
   });
+  fetchCandidateMap.mockImplementationOnce(() => {
```

Comment:
If you don't have this mock added, then when you call `fetchCandidateMap`, it returns `undefined`, which trips people up. Are we testing people's knowledge of jest.

Reply:
Again not sure if this is intentional, I guess we need someone from News team who originally came up with this to clarify.

Reply:
When I look at it, the first one looks like a mistake we should correct, and the second one is part of the implementation I would expect from the candidate - I wouldn't mock something that's not needed in the tests as given to the candidate.  I'll check with Craig.

Reply:
This is a really good one that I can imagine going both ways on.

I agree that we probably shouldn't add mocks that are not used in the test itself as that could be confusing in itself.

I _think_ it's useful here for the candidate to realise that they need to extend the mocking of the API to write tests for the candidate map.

Jests means of mocking functions isn't very different to other testing libraries, and the only part that is really bespoke is the name of `mockImplementationOnce` which can literally be copy pasted from the mock on `fetchResultData` above. I think this means we are testing their ability to use mocks in tests, rather than Jest itself, which feels valid to me, but I'll happily be challenged on that one.

Can we data drive this? How many times does this cause issues for candidates? Perhaps this is a common "tripping up point" that we need to ensure that the assessors are aware of, so they can step in if candidates get stuck with mocking specifics, particularly if the candidate has less development experience

At the moment I'm inclined to suggest that we keep this one as it is, but happy to discuss more.